pipeline-runner/R/gem2s-1-download_cellranger.R

```diff
@@ -47,13 +54,11 @@ download_cellranger <- function(input, pipeline_config) {
     config$metadata <- input$metadata
```

Comment:
Lines 36 - 39: there is no error handling around the call to AWS. If the object is not in the S3 bucket, then this part of code will crash. Before we write the output on a file (line 42), we should check that download was ok, that the files are there. If they are not there, we should raise an exception.

Reply:
Let me know if this doesn't fully address the concern but in `init.R` function `init`, every task is wrapped in a `tryCatchLog` which, if anything fails will result in the pipeline returning an error message to the API and a callstack in the logs.

Reply:
It will definitely error out at some point because of that, but looking at the code, it looks like it will error out much later after the error occurs, which will make it harder to debug. So I think what would happen if there is no such object in the bucket is:

1. Lines 36 to 39 will run successfully. There will be no error raised there. body will be either empty or undefined.
2. If body is evaluated to None/null, then line 42 will most likely run successfully. writeBin function will just write an empty content to the supplied `local_fpath`.
3. The next place where we try to read from `local_fpath` and do things on the contents of `local_fpath` will fail,  because it will be empty. This `tryCatchLog` check in `init.R` will detect it and raise an error at the place where we are trying to do things with the empty contents of `local_fpath`. 

The problem with this flow is that the error in 3. won't show that the real problem comes from 1. This will take us much more time to trace back and debug the error. This is why a good practice is to try and detect and raise issues as early as possible. 
So in this case with this code, after lines 36 to 39 I would add a check if body is empty, and if it is, I will raise an error right there.